samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -464,7 +488,8 @@ private boolean deQueueAction(ContainerPlacementRequestMessage requestMessage) {
     Boolean invalidAction = false;
     String errorMessage = null;
     if (!samzaApplicationState.runningProcessors.containsKey(requestMessage.getProcessorId()) &&
-        !samzaApplicationState.pendingProcessors.containsKey(requestMessage.getProcessorId())
+        !samzaApplicationState.pendingProcessors.containsKey(requestMessage.getProcessorId()) &&
+        !samzaApplicationState.failedProcessors.containsKey(requestMessage.getProcessorId())
```

Comment:
To make this readable, maybe define variables for each of the lookups and 
define the if to be 
if (inRunning || inPending || inFailed) {//do blah}
else {
this
}